src/blueprints/tv2/helpers/tv2-asset-path-helper.ts

```diff
@@ -0,0 +1,62 @@
+export class Tv2AssetPathHelper {
+  public joinAssetToNetworkPath(
+    networkPath: string,
+    assetFile: string,
+    extensionWithLeadingDot: string,
+    folder: string
+  ): string {
+    const folderWithForwardSlashes: string = this.replaceDoubleBackslashWithForwardSlash(folder)
+    const assetWithForwardSlashes: string = this.replaceDoubleBackslashWithForwardSlash(assetFile)
+    const networkPathWithForwardSlashes: string = networkPath[0] + this.replaceDoubleBackslashWithForwardSlash(networkPath.slice(1))
+
+    const folderWithoutLeadingTrailingSlashes = this.removeLeadingSlash(this.removeTrailingSlash(folderWithForwardSlashes))
+    const assetFileWithoutLeadingSlashes = this.removeLeadingSlash(assetWithForwardSlashes)
+    const networkPathWithoutTrailingSlashes = this.removeTrailingSlash(networkPathWithForwardSlashes)
+
+    const extension = this.removeLeadingDot(extensionWithLeadingDot)
+
+    if (!folderWithoutLeadingTrailingSlashes) {
+      return this.convertToWindowsPath(`${networkPathWithoutTrailingSlashes}/${assetFileWithoutLeadingSlashes}.${extension}`)
+    }
+
+    return this.convertToWindowsPath(`${networkPathWithoutTrailingSlashes}/${folderWithoutLeadingTrailingSlashes}/${assetFileWithoutLeadingSlashes}.${extension}`)
+  }
+
+  public joinAssetToFolder(assetFile: string, folder?: string): string {
+    if (!folder) {
+      return assetFile
+    }
+
+    const folderWithForwardSlashes: string = this.replaceDoubleBackslashWithForwardSlash(folder)
+    const assetWithForwardSlashes: string = this.replaceDoubleBackslashWithForwardSlash(assetFile)
+
+    const folderWithoutTrailingSlashes: string = folderWithForwardSlashes.replace(/\/+$/, '')
+    const assetFileWithoutLeadingSlashes: string = assetWithForwardSlashes.replace(/^\/+/, '')
+
+    return `${folderWithoutTrailingSlashes}/${assetFileWithoutLeadingSlashes}`
+  }
```

Comment:
We created the current method together. There might very well be room for improvement, in a manor as suggested, but i don't have a good enough understanding to do so by myself.

Perhabs this change should be done in a seperate PR, with or without me. I am hoping to get this PR merged sooner rather than later, as it has already lived for quite a while...

Reply:
If we are uncertain about what the code does, then it seems like a good idea to write tests for it.
Doing that will also help us understand what happens and it increases the likelihood of correct behaviour when refactoring the code. If you need help understanding the use cases, then we can look at the use cases together.

We are aiming for writing clean code. This is better than what is in the blueprints, but we have duplicated logic, and multiple responsibilities in each method. The argument that the PR is long lived is not a good reason to slack on our standards.

Reply:
The commented on code has been reworked together with Anders for better readability. In doing so we also created tests for it.